src/realm/index_string.cpp

```diff
@@ -420,6 +420,12 @@ struct SearchList {
 
 void IndexArray::index_string_all_ins(StringData value, IntegerColumn& result, ColumnBase* column) const
 {
+    if (value.is_null()) {
+        // we can't use case_map on null strings because it currently returns an
+        // empty string ("") in that case which is different than a null StringData
+        return index_string_all(value, result, column);
```

Comment:
Shouldn't it be solved by letting case_map() return null (StringData(0, 0)) then?

Reply:
@rrrlasse yes, the first fix I tried was to return an empty optional and that small change actually passes all tests. The reason I didn't stay with that is because we have many other callers that use `case_map` and they assume that null is returned only when the case mapping fails (eg. mapping malformed utf8). So we have checks in other places that would have to change to adapt to the new behaviour of returning null. I decided to localise the fix to only the case insensitive code to reduce the potential impact of introducing new bugs in other places.